src/main/java/edu/utdallas/davisbase/storage/IndexPage.java

```diff
@@ -0,0 +1,497 @@
+package edu.utdallas.davisbase.storage;
+
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IndexPage {
+
+	static final int BYTES_OF_PAGE_OFFSET = Short.BYTES;
+	static final int PAGE_OFFSET_OF_CELL_COUNT = 0x01;
+	static final int PAGE_OFFSET_OF_RIGHTMOST_PAGE_NO = 0x06;
+	static final int PAGE_OFFSET_OF_CELL_PAGE_OFFSET_ARRAY = 0x10;
+	static final int PAGE_OFFSET_OF_CELL_CONTENT_START_POINT = 0x03;
+	static final int PAGE_OFFSET_OF_CELL_PARENT_PAGE_NO = 0x0A;
+	static final int pageSize = StorageConfiguration.Builder.getDefaultPageSize();
+	static final int maximumNoOFKeys = 2;
+	static final long metaDataRootPageNoOffsetInFile = 0x05;
+
+	private static int AddInteriorPage(RandomAccessFile file) {
+		int numofPages = 0;
+		try {
+			numofPages = (int) (file.length() / pageSize);
+			numofPages = numofPages + 1;
+			file.setLength(pageSize * numofPages);
+			file.seek((numofPages - 1) * pageSize);
+			file.writeByte(0x02);// writing page type
+			file.seek(((numofPages - 1) * pageSize) + PAGE_OFFSET_OF_RIGHTMOST_PAGE_NO);
+			file.writeInt(-1); // setting right most child to -1
+		} catch (Exception e) {
+			System.out.println(e);
+		}
+		return numofPages;
+	}
+
+	private static int AddLeafPage(RandomAccessFile file) {
+		int numofPages = 0;
+		try {
+			numofPages = (int) (file.length() / pageSize);
+			numofPages = numofPages + 1;
+			file.setLength(pageSize * numofPages);
+			file.seek((numofPages - 1) * pageSize);
+			file.writeByte(0x0A);// writing page type
+
+			file.seek(((numofPages - 1) * pageSize) + 6);
+			file.writeInt(-1);// setting right most child to -1
+
+		} catch (Exception e) {
+			System.out.println(e);
+		}
+		return numofPages;
+	}
+
+	public static void addTableMetaDataPage(RandomAccessFile file) throws IOException {
+		try {
+			file.setLength(pageSize);
+			file.seek(0x00);
+			file.writeByte(0);
+			int firstPageNo = AddLeafPage(file);
+			file.seek(0x05);
+			file.writeInt(firstPageNo); // writing root page no.
+			setPageasRoot(file, firstPageNo);
+		} catch (Exception e) {
+
+		}
+	}
+
+	private static void setPageasRoot(RandomAccessFile file, int pageNo) {
+		// TODO Auto-generated method stub
+		int seekParentByte = (pageNo - 1) * pageSize + 10;
+		try {
+			file.seek(seekParentByte);
+			file.writeInt(-1);// making root
+			updateMetaDataRoot(file, pageNo);
+		} catch (Exception e) {
+		}
+	}
+
+	private static void updateMetaDataRoot(RandomAccessFile file, int pageNo) throws IOException {
+		// TODO Auto-generated method stub
+		file.seek(metaDataRootPageNoOffsetInFile);
+		file.writeInt(pageNo);
+	}
+
+	// splitting leaf page
+	public static int splitLeafPage(RandomAccessFile file, int pageNo) throws IOException {
+		sortKeys(file, pageNo);
+		int newSiblingPageNo = AddLeafPage(file);
+		boolean rootflag = CheckifRootNode(file, pageNo);
```

Comment:
Use "camelCasing" for local variables.

```suggestion
		boolean rootFlag = CheckifRootNode(file, pageNo);
```